libats-slick/src/test/scala/com/advancedtelematic/libats/slick/db/SlickExtensionsSpec.scala

```diff
@@ -42,4 +43,29 @@ class SlickExtensionsSpec extends FunSuite with Matchers with ScalaFutures with
 
     f.futureValue shouldBe Book(10, "Some book")
   }
+
+  test("resultHead on a Query returns the error in arg") {
+    val f = db.run(books.filter(_.id === 20l).resultHead(Error))
+    f.failed.futureValue shouldBe Error
+  }
+
+  test("maybeFilter uses filter if condition is defined") {
+    val f = for {
+      _ <- db.run(books += Book(30, "Some book", Option("30 some code")))
+      result <- db.run(books.maybeFilter(_.id === Option(30l)).result)
```

Comment:
So I suppose this is one of those situations that it would be good to use an ide, because I can't figure out how this works. But if you compare a `x : Rep[T]` with an `Option[T]` on we get a `Rep[Option[Boolean]]`?